spec/controllers/api/v1/tokens_controller_spec.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe API::V1::TokensController, type: :controller do
+  describe 'Filters' do
+    it { is_expected.not_to use_before_action(:doorkeeper_authorize!) }
+  end
+
+  describe 'POST#revoke' do
+    context 'given a valid access token' do
+      it 'returns success status' do
+        Fabricate(:access_token, token: 'access_token')
+
+        post :revoke, params: { token: 'access_token' }
+
+        expect(response).to have_http_status(:success)
+      end
+
+      it 'returns an empty response' do
+        Fabricate(:access_token, token: 'access_token')
+
+        post :revoke, params: { token: 'access_token' }
+
+        expect(JSON.parse(response.body)).to be_empty
+      end
+    end
+
+    context 'given a valid refresh token' do
+      it 'returns success status' do
+        Fabricate(:access_token, refresh_token: 'refresh_token')
+
+        post :revoke, params: { token: 'refresh_token' }
+
+        expect(response).to have_http_status(:success)
+      end
+
+      it 'returns an empty response' do
+        Fabricate(:access_token, refresh_token: 'refresh_token')
+
+        post :revoke, params: { token: 'refresh_token' }
+
+        expect(JSON.parse(response.body)).to be_empty
+      end
+    end
+
+    context 'given an invalid token' do
+      it 'returns success status' do
+        post :revoke, params: { token: 'invalid' }
+
+        expect(response).to have_http_status(:success)
```

Comment:
So there is no validation at all? Also, should we have a test with no param `token` sent?

Reply:
Yes, doorkeeper just checks the authorization application, but we removed it, then it returns success for all `revoke` request

Reply:
Very useful endpoint 😆